### PR TITLE
save smart contract template  params in ASCCache

### PR DESCRIPTION
**Pull request by rmeena840**
save smart contract template  params in ASCCache

### CODE REVIEW

File: packages/algob/src/lib/compile.ts [hunk 1]
```diff
@@ -113,7 +116,7 @@ export class CompileOp {
 		return this.algocl.compile(code).do();
 	}
 
-	async compile(filename: string, tealCode: string, tealHash: number): Promise<ASCCache> {
+	async compile(filename: string, tealCode: string, tealHash: number, SCParams: SCParams): Promise<ASCCache> {
```
Comment (ratik21) [inline packages/algob/src/lib/compile.ts:119]: + Naming convention is wrong (`SCParams`). 
+ Also, shouldn't `scParams` be an optional parameter here?

File: packages/algob/test/builtin-tasks/compile.ts [hunk 2]
```diff
@@ -34,6 +34,7 @@ class CompileOpMock extends CompileOp {
 			srcHash: tealHash,
 			tealCode: _tealCode,
 			base64ToBytes: new Uint8Array(1),
+			SCParams: SCParams
```
Comment (ratik21) [inline packages/algob/test/builtin-tasks/compile.ts:37]: Can we add an assertion, somewhere in the test to verify *scParams*?

**Review by ratik21 — COMMENTED** [review]
Changelog needs to be updated

File: packages/algob/test/fixture-projects/config-project/assets/gold-asa-py-check.yaml [hunk 3]
```diff
@@ -35,3 +35,4 @@ tealCode: |
   return
 base64ToBytes:
   - 0
+scParams: {}
```
Comment (vuvth) [inline packages/algob/test/fixture-projects/config-project/assets/gold-asa-py-check.yaml:38]: I not sure yaml support to save JSON object like this. Can you check it?

File: packages/algob/test/builtin-tasks/compile.ts [hunk 4]
```diff
@@ -127,6 +128,7 @@ describe("Compile task", () => {
 	it("should return correct ASCCache from CompileOp", async () => {
```
Comment (vuvth) [inline packages/algob/test/builtin-tasks/compile.ts:128]: Could you add test suit follow this suit? 
1. compile and ASACache to cache file. 
2. Load this file and verify we stored scParams.

**Review by vuvth — COMMENTED** [review]
Thank for your PR! I left few comments.

**Review by sczembor — COMMENTED** [review]
The `SCParams` should be changed to `scParams`

**Review by vuvth — APPROVED** [review]
LGTM!

**Review by robert-zaremba — APPROVED** [review]
changelog update is missing.

**Review by ratik21 — APPROVED** [review]

**Pull request merged**